lib/checks/label/label-content-name-mismatch.js

```diff
@@ -0,0 +1,25 @@
+const {
+	text: {
+		accessibleText,
+		visible // get visible text
+	}
+} = axe.commons;
+
+/**
+ * Note:
+ * `label-content-name-mismatch-matches` ignore `node`
+ * if there is no `accessibleText` or `textContent`
+ */
+const accText = accessibleText(node).toLowerCase();
+const visibleTextContent = visible(node).toLowerCase();
+
+/**
+ * if `textContent` is not part of `accessibleText` -> fail
+ */
+
+if (!accText.includes(visibleTextContent)) {
```

Comment:
How about `<button aria-label="next page">next page &gt;</button>`, or things similar. I think we need to come up with a way to handle things like punctuations and other things that people wouldn't "read out". Another scenario is `<button aria-label="page 1">page one</button>` or the inverse of it.